src/lib/frontend/cnf.ml

```diff
@@ -116,12 +116,13 @@ let rec make_term up_qv quant_basename t =
 
     | TTlet (binders, t2) ->
       let binders =
-        List.rev_map (fun (s, t1) -> s, mk_term t1)
+        List.rev_map (fun (s, t1) ->
+            match s with Sy.Var v -> v, mk_term t1 | _ -> assert false)
```

Comment:
Actually I'm checking it right now. I forgot to do it before publishing the PR. I hope the legacy parser enforce this invariant (the new one does).

Reply:
Actually, before submitting this PR, I tried to change the type of the binders in the legacy typechecker. This will requires to change many types, mostly because the typechecker aggregates the symbols in a context, so embed variables into symbols makes the design simpler here. I decided to keep the code as before. The only place where we produce a new `TTlet` term is here:
https://github.com/OCamlPro/alt-ergo/blob/7363cd05a0b3af10a6a6b98738ee0810899052b2/src/lib/frontend/typechecker.ml#L913-L934

As you can see line 926, we store new variables in the environment of the typechecker and we retrieve them at the line 932 to produce the list of binders.

Reply:
I suggest you add to the assert false a TODO stating that we could actually simplify but it would require to update some legacy code which was out of the scope of the PR adding this assert false (and a reference to typechecker.ml).